packages/repository/src/relations/has-many/has-many-through.helpers.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright IBM Corp. 2018,2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as debugFactory from 'debug';
+import {DataObject} from '../../common-types';
+import {camelCase} from 'lodash';
+import {Entity} from '../../model';
+import {InvalidRelationError} from '../../errors';
+import {isTypeResolver} from '../../type-resolver';
+import {HasManyThroughDefinition} from '../relation.types';
+
+const debug = debugFactory('loopback:repository:has-many-through-helpers');
+
+/**
+ * Relation definition with optional metadata (e.g. `keyTo`) filled in.
+ * @internal
+ */
+export type HasManyThroughResolvedDefinition = HasManyThroughDefinition & {
+  keyTo: string;
+  keyThrough: string;
+  targetPrimaryKey: string;
+};
+
+/**
+ * Creates constraint used to query target
+ * @param relationMeta - hasManyThrough metadata to resolve
+ * @param throughInstances - Instances of through entities used to constrain the target
+ * @internal
+ */
+export function createTargetConstraint<
+  Target extends Entity,
+  Through extends Entity
+>(
+  relationMeta: HasManyThroughResolvedDefinition,
+  throughInstances: Through[],
+): DataObject<Target> {
+  const {targetPrimaryKey} = relationMeta;
+  const targetFkName = relationMeta.keyThrough;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const constraint: any = {
+    or: throughInstances.map((throughInstance: Through) => {
+      return {
+        [targetPrimaryKey]: throughInstance[targetFkName as keyof Through],
```

Comment:
I chose to fix with `inq` instead of using `findByForeignKeys` because it keeps the `constraint` logic consistent.

f72b958e98d6b074e437c18b95ee045571545b7b